CMakeLists.txt

```diff
@@ -346,3 +345,23 @@ if(ENABLE_TESTING)
                     ${CMAKE_CURRENT_BINARY_DIR}/DartConfiguration.tcl COPYONLY)
     endif()
 endif()
+
+include(CMakePackageConfigHelpers)
+
+write_basic_package_version_file(
+    "cmake/MbedTLSConfigVersion.cmake"
+    COMPATIBILITY AnyNewerVersion
+    VERSION 2.26.0)
+
+configure_package_config_file(
+    "cmake/MbedTLSConfig.cmake.in"
+    "cmake/MbedTLSConfig.cmake"
+    INSTALL_DESTINATION "cmake/MbedTLSConfig.cmake")
+
+install(
+    EXPORT MbedTLSTargets
+    DESTINATION "cmake"
+    FILE "MbedTLSTargets.cmake")
+
+# Make this package visible to the system
+export(PACKAGE MbedTLS)
```

Comment:
I think it is better to avoid using the "user package registry". This is a "global" setting and the project can not really know how the integrator wishes to use it, and what side effects there can be. I don't know how this works with cross-compiled packages or even if there is any well defined cmake methodology to handle this.

Reply:
[The behaviour around this](https://cmake.org/cmake/help/latest/policy/CMP0090.html#policy:CMP0090) has changed a lot over the versions, flipping between being enabled by default and disabled by default. As we're targeting CMake 2.8.12 at minimum I think the policy is "enabled by default", so what I can do is only only enable this line with CMake 3.15+, where global package exporting is disabled by default - this at least gives the user the option to export the package if they so desire with [`CMAKE_EXPORT_PACKAGE_REGISTRY`](https://cmake.org/cmake/help/latest/variable/CMAKE_EXPORT_PACKAGE_REGISTRY.html#variable:CMAKE_EXPORT_PACKAGE_REGISTRY).

Reply:
I see. That sounds like a good workaround.